skylight-hq/phinvads-go
# Update USA Banner component

resolves #33 

## What this PR changes:
- adds a new route to enable `htmx` swapping for show/hide functionality on the USA banner dropdown content
  - updates templ template to accept a parameter for showing/hiding the banner contents
- updates "Here's how you know" clickable element:
  - replaces the `<a>` element with a `<button>` element for better keyboard navigability
- updates styles to match the banner at [https://www.cdc.gov/phin/php/vocabulary/](https://www.cdc.gov/phin/php/vocabulary)

## How to test:
- Test at both full browser screen width (> 1000px) and smaller responsive widths ( < 900px)
- Click "Here's how you know" and make sure the content properly shows/hides and styles do not change
- Make sure you can interact with the clickable element with both a mouse/trackpad click, as well as tabbing to it and clicking with the space/enter key

## Notes:
- Keyboard navigability needs additional work; when using the keyboard to show/hide the banner content, focus should remain on the button element; currently you need to tab backwards to be able to focus and click it again. I'll add another [ticket](https://github.com/skylight-hq/phinvads-go/issues/39) for this.

## Feedback requested:
- ~How can we use the `Templ` templating instead of creating additional HTML files to use with HTMX functionality?~ addressed in [99cbe18](https://github.com/skylight-hq/phinvads-go/pull/37/commits/99cbe1879d5fb926c8051d0c724957fa458e84f3)
